sources/sqlserver/infoschema.go

```diff
@@ -0,0 +1,383 @@
+		if err != nil {
+			conv.Unexpected(fmt.Sprintf("Can't scan: %v", err))
+			continue
+		}
+		ignored := schema.Ignored{}
+		for _, c := range constraints[colName] {
+			// c can be UNIQUE, PRIMARY KEY, FOREIGN KEY,
+			// or CHECK (based on msql, sql server, postgres docs).
+			// We've already filtered out PRIMARY KEY.
+			switch c {
+			case "CHECK":
+				ignored.Check = true
+			case "FOREIGN KEY", "PRIMARY KEY", "UNIQUE":
+				// Nothing to do here -- these are handled elsewhere.
+			}
+		}
+		ignored.Default = colDefault.Valid
+		c := schema.Column{
+			Name:    colName,
+			Type:    toType(dataType, charMaxLen, numericPrecision, numericScale),
+			NotNull: strings.ToUpper(isNullable) == "NO",
+			Ignored: ignored,
+		}
+		colDefs[colName] = c
+		colNames = append(colNames, colName)
+	}
+	return colDefs, colNames, nil
+}
+
+// GetConstraints returns a list of primary keys and by-column map of
+// other constraints.  Note: we need to preserve ordinal order of
+// columns in primary key constraints.
+// Note that foreign key constraints are handled in getForeignKeys.
+func (isi InfoSchemaImpl) GetConstraints(conv *internal.Conv, table common.SchemaAndName) ([]string, map[string][]string, error) {
+	q := `
+		SELECT 
+			k.COLUMN_NAME, 
+			t.CONSTRAINT_TYPE
+		FROM INFORMATION_SCHEMA.TABLE_CONSTRAINTS AS t
+		INNER JOIN INFORMATION_SCHEMA.KEY_COLUMN_USAGE AS k 
+			ON t.CONSTRAINT_NAME = k.CONSTRAINT_NAME AND t.CONSTRAINT_SCHEMA = k.CONSTRAINT_SCHEMA
+		WHERE k.TABLE_SCHEMA = @p1 AND k.TABLE_NAME = @p2 ORDER BY k.ordinal_position;
+	`
+	rows, err := isi.Db.Query(q, table.Schema, table.Name)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer rows.Close()
+	var primaryKeys []string
+	var col, constraint string
+	m := make(map[string][]string)
+	for rows.Next() {
+		err := rows.Scan(&col, &constraint)
+		if err != nil {
+			conv.Unexpected(fmt.Sprintf("Can't scan: %v", err))
+			continue
+		}
+		if col == "" || constraint == "" {
+			conv.Unexpected(fmt.Sprintf("Got empty col or constraint"))
+			continue
+		}
+		switch constraint {
+		case "PRIMARY KEY":
+			primaryKeys = append(primaryKeys, col)
+		default:
+			m[col] = append(m[col], constraint)
+		}
+	}
+	return primaryKeys, m, nil
+}
+
+// GetForeignKeys returns a list of all the foreign key constraints.
+func (isi InfoSchemaImpl) GetForeignKeys(conv *internal.Conv, table common.SchemaAndName) (foreignKeys []schema.ForeignKey, err error) {
+	q := `
+	SELECT 
+		OBJECT_SCHEMA_NAME (FK.referenced_object_id) AS [schema_name],
+		OBJECT_NAME (FK.referenced_object_id) AS [referenced_table],
+		COL_NAME(FKC.parent_object_id, FKC.parent_column_id) AS [column],  
+		COL_NAME(FKC.referenced_object_id, FKC.referenced_column_id) AS [referenced_column],  
+		FK.name AS [foreign_key_name]
+	FROM sys.foreign_keys AS FK  
+	INNER JOIN sys.foreign_key_columns AS FKC   
+    ON FK.object_id = FKC.constraint_object_id  
+	WHERE FK.parent_object_id = OBJECT_ID(@p1);
```

Comment:
Im unsure what OBJECT_ID is, but I see table.Schema goes as an input to it. If we are not using table.Name here since I dont see a @p2 in the query, we can prolly remove it from line 264. Similarly in other queries (if any)

Reply:
-OBJECT_ID returns the id of the database object, in this case, the table that is passed in @p1
-Actually, only one argument is passed in the next statement. It is formatted like "<schma name>.<table name>". The query will fetch the detail for the required schema and table only.

Reply:
Ah right my bad, missed the fmt.Sprintf